lib/src/block/rle.rs

```diff
@@ -1,7 +1,11 @@
-pub fn rle(input: &[u8]) -> Vec<u8> {
+pub fn rle(
+    input: &[u8],
+    init_counter: usize,
+    init_last: Option<u8>,
+) -> (Vec<u8>, usize, Option<u8>) {
```

Comment:
Does it make sense to return a struct here naming the purpose of the components?

Reply:
Done. https://github.com/torfmaster/ribzip2/pull/17/files#diff-d885fa576bdfc7c227b9e80f8c26e58da04a194466ae8b67f65791c21dfd6e4aR1-R5